### DISCUSSION TITLE
Readonly data

### DISCUSSION THREAD

**Discussion by ondrej-langr** [local-first-web/auth#137] [General]
Hello, i love the idea of local first and started experimenting with some libs that go with the idea of localfirst, but this seems most complete as far as i can tell. Mostly as this project does not discredit the need of colaboration (roles, authorization, etc)

I get around mostly with reading the provided documentation and code, but what I still dont see is how to create readonly data. Lets say user creates post, but only specific user roles and the owner can update or delete it for example. This would be hardly enforceable on the client side(?), but duable when synchronizing.

Is there a way to achieve this ATM or is that not a option?

From what I get, secure data can be shared between specific user roles (and probably with specific users with lockboxes), but i dont see a way to restrict users from editing the data between each users (Alice can edit Bobs post and even declare it their "own" and viceversa)

Happy to hear your ideas and help

**Comment by JasonKleban** [comment]
Yes, I am just starting to explore this neat library along with Automerge.  I am interested in a technique for creating a distributed document/database where some of the replicas are, though not enforceable locally, read-only from the perspective of the replica peers. Any replicas in a mesh should be able to forward updates appropriately signed, but not introduce changes or unrecoverable corruption.

> Alice can edit Bobs post and even declare it their "own" and viceversa

We could detect invalid data through in-document signatures, but by the time they're consumed into the document locally for verification, there's no obvious way to recover the unforged document state.  Hashes of the resulting snapshot won't do because replicas are only _eventually_ consistent.
